File: android/src/main/kotlin/com/benjamin/horner/helpers/CountryCodeHelper.kt
package com.benjamin.horner.flutter_acs_card_reader

import android.util.Log

class CountryCodeHelper {
    fun handleCountryCode(hex: String): String {
        val ccHex: String = hex.substring(0, 2)
        Log.e("handleCountryCode", "$ccHex")
        return hexToCountryCode(ccHex)
    }

    private fun hexToCountryCode(hex: String): String {
        return when (Integer.parseInt(hex, 16)) {
            0 -> "___"
            1 -> "A__"
            2 -> "AL_"
            3 -> "AND"
            4 -> "ARM"
            5 -> "AZ_"
            6 -> "B__"
            7 -> "BG_"
            8 -> "BIH"
            9 -> "BY_"
            10 -> "CH_"
            11 -> "CY_"
            12 -> "CZ_"
            13 -> "D__"
            14 -> "DK_"
            15 -> "E__"
            16 -> "EST"
            17 -> "F__"
            18 -> "FIN"
            19 -> "FL_"
            20 -> "FR_"
            21 -> "UK_"
            22 -> "GE_"
            23 -> "GR_"
            24 -> "H__"
            25 -> "HR_"
            26 -> "I__"
            27 -> "IRL"
            28 -> "IS_"
            29 -> "KZ_"
            30 -> "L__"
            31 -> "LT_"
            32 -> "LV_"
            33 -> "M__"
            34 -> "MC_"
            35 -> "MD_"
            36 -> "MK_"
            37 -> "N__"
            38 -> "NL_"
            39 -> "P__"
            40 -> "PL_"
            41 -> "RO_"
            42 -> "RSM"
            43 -> "RUS"
            44 -> "S__"
            45 -> "SK_"
            46 -> "SLO"
            47 -> "TM_"
            48 -> "TR_"
            49 -> "UA_"
            50 -> "V__"
            51 -> "YU_"
            253 -> "EC_"
            254 -> "EUR"
            255 -> "WLD"
            else -> "UNK"
        }
    }
}

File: ios/Classes/Helpers/CountryCodeHelper.swift
//
//  CountryCodeHelper.swift
//  flutter_acs_card_reader
//
//  Created by Benjamin Horner on 23/11/2023.
//

import Foundation

class CountryCodeHelper {
    func handleCountryCode(hex: String) -> String {
        let ccHex = String(hex.prefix(2))
        print("handleCountryCode \(ccHex)")
        return hexToCountryCode(ccHex)
    }

    private func hexToCountryCode(_ hex: String) -> String {
        guard let intValue = Int(hex, radix: 16) else {
            return "UNK"
        }

        switch intValue {
        case 0: return "___"
        case 1: return "A__"
        case 2: return "AL_"
        case 3: return "AND"
        case 4: return "ARM"
        case 5: return "AZ_"
        case 6: return "B__"
        case 7: return "BG_"
        case 8: return "BIH"
        case 9: return "BY_"
        case 10: return "CH_"
        case 11: return "CY_"
        case 12: return "CZ_"
        case 13: return "D__"
        case 14: return "DK_"
        case 15: return "E__"
        case 16: return "EST"
        case 17: return "F__"
        case 18: return "FIN"
        case 19: return "FL_"
        case 20: return "FR_"
        case 21: return "UK_"
        case 22: return "GE_"
        case 23: return "GR_"
        case 24: return "H__"
        case 25: return "HR_"
        case 26: return "I__"
        case 27: return "IRL"
        case 28: return "IS_"
        case 29: return "KZ_"
        case 30: return "L__"
        case 31: return "LT_"
        case 32: return "LV_"
        case 33: return "M__"
        case 34: return "MC_"
        case 35: return "MD_"
        case 36: return "MK_"
        case 37: return "N__"
        case 38: return "NL_"
        case 39: return "P__"
        case 40: return "PL_"
        case 41: return "RO_"
        case 42: return "RSM"
        case 43: return "RUS"
        case 44: return "S__"
        case 45: return "SK_"
        case 46: return "SLO"
        case 47: return "TM_"
        case 48: return "TR_"
        case 49: return "UA_"
        case 50: return "V__"
        case 51: return "YU_"
        case 253: return "EC_"
        case 254: return "EUR"
        case 255: return "WLD"
        default: return "UNK"
        }
    }
}

